jax_distribution/types.py

```diff
@@ -43,6 +43,18 @@ class LearnerState(NamedTuple):
     timestep: TimeStep
 
 
+class RNNLearnerState(NamedTuple):
+    """State of the `Learner` for recurrent architectures."""
+
+    params: FrozenDict
+    opt_state: OptState
+    key: chex.PRNGKey
+    env_state: LogEnvState
+    timestep: TimeStep
+    dones: chex.Array
+    hstate: chex.Array
```

Comment:
Can we make `LearnerState` a `chex.dataclass` and then 
```suggestion
@chex.dataclass
class RNNLearnerState(LearnerState):
    dones: chex.Array
    hstate: chex.Array
```